geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/FeatureUtils.scala

```diff
@@ -116,7 +116,11 @@ object FeatureUtils {
     copyToFeature(writer.next(), sf, useProvidedFid)
 
   def copyToFeature(toWrite: SimpleFeature, sf: SimpleFeature, useProvidedFid: Boolean = false): SimpleFeature = {
-    toWrite.setAttributes(sf.getAttributes)
+    var i = 0
+    while (i < sf.getAttributeCount) {
+      toWrite.setAttribute(i, sf.getAttribute(i))
```

Comment:
Does this skip the convert call?

Reply:
no this will still run the convert. it will just avoid allocating a new list in the call to getAttributes()